sortinghat/core/schema.py

```diff
@@ -77,6 +82,61 @@ def convert_json_field_to_generic_scalar(field, registry=None):
     return OperationArgsType(description=field.help_text, required=not field.null)
 
 
+def parse_date_filter(filter_value):
+    """Extract the filter terms from a date filter
+
+    The accepted formats are controlled by regular expressions
+    matching two patterns: a comparison operator (>, >=, <, <=) and a date
+    OR a range operator (..) between two dates.
+
+    The accepted date format is ISO 8601, YYYY-MM-DDTHH:MM:SS, also
+    accepting microseconds and time zone offset (YYYY-MM-DDTHH:MM:SS.ms+HH:HH).
+
+    :param filter_value: String containing the filter value
+
+    :returns: A dictionary including an operator and the date values
+    """
+    # Accepted date format is ISO 8601, YYYY-MM-DDTHH:MM:SS
+    filter_data = {
+        "operator": None,
+        "date1": None,
+        "date2": None
+    }
+
+    iso_date_group = "(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?(\+\d{2}:\d{2})?)"
+    # Filter with a comparison operator (>, >=, <, <=) and a date (e.g. `>=YYYY-MM-DDTHH:MM:SS`)
+    oper_comparison = r"^(<=?|>=?)%s$" % iso_date_group
+    # Filter with a range operator (..) between two dates
+    # (e.g. YYYY-MM-DDTHH:MM:SS..YYYY-MM-DDTHH:MM:SS)
+    range_comparison = r"^%s\.{2}%s$" % (iso_date_group, iso_date_group)
+
+    oper_result = re.match(oper_comparison, filter_value)
+    range_result = re.match(range_comparison, filter_value)
+
+    if not oper_result and not range_result:
+        raise InvalidFilterError(msg='Filter format is not valid')
+
+    if oper_result:
+        filter_data['operator'] = oper_result.group(1)
+        try:
+            filter_data['date1'] = str_to_datetime(oper_result.group(2))
+        except InvalidDateError as e:
+            raise InvalidFilterError(msg=e)
+
+    if range_result:
+        filter_data['operator'] = '..'
+        try:
+            filter_data['date1'] = str_to_datetime(range_result.group(1))
+            filter_data['date2'] = str_to_datetime(range_result.group(4))
+        except InvalidDateError as e:
+            raise InvalidFilterError(msg=e)
+
+        if filter_data['date1'] > filter_data['date2']:
+            raise InvalidFilterError(msg='Date range is invalid, date 2 must be greater than date 1')
```

Comment:
Instead of saying `date 2` and `date 1`, why don't we say `Date range is invalid. Upper bound must be greater than the lower bound", or `Date range is invalid. First date must be greater than the second one". Let's try to find something clear.